opensea-operator-filter/test/ERC721WithOperatorFilter.sol

```diff
@@ -0,0 +1,31 @@
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import { BigNumber } from "ethers";
+
+import {
+  ERC721WithOperatorFilterer,
+  ERC721WithOperatorFilterer__factory
+
+} from "../typechain-types";
+
+
+describe("ERC721WithOperatorFilter", function () {
+  let owner: SignerWithAddress;
+  let nftUser1: SignerWithAddress;
+  let nftUser2: SignerWithAddress;
+  let nftUser3: SignerWithAddress;
+  let nftUser4: SignerWithAddress;
+  let nftUser5: SignerWithAddress;
+
+  let erc721WithOperatorFilterer: ERC721WithOperatorFilterer;
+
+  before(async function () {
+    [owner, nftUser1, nftUser2, nftUser3, nftUser4] = await ethers.getSigners();
+
+    const erc721WithOperatorFilterFactory = new ERC721WithOperatorFilterer__factory(owner)
+
+    erc721WithOperatorFilterer = await erc721WithOperatorFilterFactory.deploy("Test NFT token", "NFT");
+
+  })
+});
```

Comment:
I would recommend adding more test cases, i.e. elaborate why we need OperatorFilter, and if we register the operator, what can we do, what can't we do.
Through the test case, you should address how this OperatorFilter restricts transfer, how OpenSea can enforce creator fee, etc.

Reply:
I used foundry framework to test the contract. 
that worked very well.